src/Evolution/DiscontinuousGalerkin/ProjectToBoundary.hpp

```diff
@@ -0,0 +1,227 @@
+                                  number_of_independent_components},
+                   volume_mesh.extents());
+  } else {
+    ASSERT(Mesh<Dim>(volume_mesh.extents(0), volume_mesh.basis(0),
+                     Spectral::Quadrature::GaussLobatto) == volume_mesh,
+           "The current implementation assumes the mesh be either a uniform "
+           "Gauss or Gauss-Lobatto mesh, but got "
+               << volume_mesh << ".");
+    const size_t sliced_dim = direction.dimension();
+    const size_t fixed_index = direction.side() == Side::Upper
+                                   ? volume_mesh.extents(sliced_dim) - 1
+                                   : 0;
+
+    const size_t interface_grid_points =
+        volume_mesh.extents().slice_away(sliced_dim).product();
+    const size_t volume_grid_points = volume_mesh.number_of_grid_points();
+
+    const double* vars_data = volume_fields.data();
+    // Since the face fields are a superset of the volume tags we need to find
+    // the first volume tag on the face and get the pointer for that.
+    double* interface_vars_data = get<first_volume_tag>(*face_fields)[0].data();
+
+    // The reason we can't use data_on_slice is because the volume and face tags
+    // are not the same, but data_on_slice assumes they are. In general, this
+    // function should replace data_on_slice in the long term since in
+    // additional to supporting different volume and face tags, it also supports
+    // Gauss and Gauss-Lobatto points.
+    //
+    // Run the SliceIterator as the outer-most loop since incrementing the slice
+    // iterator is surprisingly expensive.
+    for (SliceIterator si(volume_mesh.extents(), sliced_dim, fixed_index); si;
+         ++si) {
+      for (size_t i = 0; i < number_of_independent_components; ++i) {
+        // clang-tidy: do not use pointer arithmetic
+        interface_vars_data[si.slice_offset() +                      // NOLINT
+                            i * interface_grid_points] =             // NOLINT
+            vars_data[si.volume_offset() + i * volume_grid_points];  // NOLINT
+      }
+    }
+  }
+}
+
+/*!
+ * \brief Projects a subset of the tensor in the `volume_fields` onto the face
+ *
+ * The tensor to project are listed in the `TagsToProjectList`.
+ *
+ * \note This function works for both Gauss and Gauss-Lobatto uniform meshes.
+ */
+template <typename TagsToProjectList, typename VolumeVarsTagsList,
+          typename FaceVarsTagsList, size_t Dim>
+void project_tensors_to_boundary(
+    const gsl::not_null<Variables<FaceVarsTagsList>*> face_fields,
+    const Variables<VolumeVarsTagsList>& volume_fields,
+    const Mesh<Dim>& volume_mesh, const Direction<Dim>& direction) noexcept {
+  static_assert(tmpl::size<VolumeVarsTagsList>::value != 0,
+                "Must have non-zero number of volume fields");
+  static_assert(tmpl::size<FaceVarsTagsList>::value != 0,
+                "Must have non-zero number of face fields");
```

Comment:
Consider adding a static assert that each `TagsToProjectList` is in both other lists